src/js/pouch-express.js

```diff
@@ -0,0 +1,288 @@
+    that.databaseInstances[dbKey] = dbComponent;
+
+    return initPromise;
+};
+
+/**
+ *
+ * Pass along any requests to our instance of express-pouchdb.
+ *
+ * @param that - The component itself.
+ * @param req {Object} - The Express request object.
+ * @param res {Object} - The Express response object.
+ * @param next {Function} - The next piece of middleware in the chain.
+ *
+ */
+gpii.pouch.express.middleware = function (that, req, res, next) {
+    // fluid.log("express pouchdb instance '", that.id, "' responding...")
+    that.expressPouchdb(req, res, next);
+};
+
+/**
+ *
+ * Clean up any instance of `gpii.pouchdb` that we're aware of.  Then, to get rid of databases created by
+ * express-pouchdb, complete remove all data in `options.baseDir`.
+ *
+ * @param that - The component itself.
+ * @returns {Promise} - A promise that will be resolved when cleanup is complete.
+ *
+ */
+gpii.pouch.express.cleanup = function (that) {
+    // fluid.log("express pouchdb instance '", that.id, "' cleaning up...");
+    var promises = [];
+
+    fluid.each(that.databaseInstances, function (databaseInstance) {
+        promises.push(databaseInstance.destroyPouch());
+    });
+
+    var sequence = fluid.promise.sequence(promises);
+    sequence.then(function () {
+        // We cannot simply call db.destroy() on all databases, because express-pouchdb creates a few databases on its own
+        // and gives us no way to clean those up.
+        //
+        // The surest approach is to completely remove the underlying directory.  However, express-pouchdb will have one
+        // or more of those files open while it's running.
+        //
+        // Since express-pouchdb does not provide any means to kill itself, we tell it to use an alternate instance of pouch
+        // that stores its content in memory.  That way we can remove the content from the "real" directory.
+        // TODO: Review with Antranig.
+        var tmpPouchDB = PouchDB.defaults({ db: memdown});
+        that.expressPouchdb.setPouchDB(tmpPouchDB).then(function () {
+            rimraf(that.options.baseDir, function (error) {
+                if (error) {
+                    fluid.fail(error);
+                }
+                else {
+                    that.events.onCleanupComplete.fire()
+                }
+            });
+        });
+    });
+
+    return sequence;
+};
+
+fluid.defaults("gpii.pouch.express.base", {
+    gradeNames: ["fluid.component", "gpii.express.middleware"],
+    method: "use", // We have to support all HTTP methods, as does our underlying router.
+    path: "/",
+    namespace: "pouch-express", // Namespace to allow other routers to put themselves in the chain before or after us.
+    tmpDir:  os.tmpdir(),
+    baseDir: "@expand:path.resolve({that}.options.tmpDir, {that}.id)",
+    expressPouchConfigFilename: "config.json",
+    expressPouchConfigPath:     "@expand:path.resolve({that}.options.baseDir, {that}.options.expressPouchConfigFilename)",
```

Comment:
Not sure why this would be any more terrifying than calling a jqUnit function from a declarative block.  If it's a risk, I suppose I could make a small function, but it seems a waste.